### DISCUSSION TITLE
Enforcing specific cases

### DISCUSSION THREAD

**Discussion by andno037** [Cornutum/tcases#167] [Q&A] [answered]
Hi, I was just wondering if any one could point me in a good direction :)
Have tried finding the answer here http://www.cornutum.org/tcases/docs/Tcases-Guide.htm
but I can't really figure out how I can add restrictions to project-Input.xml in order to be certain that one specific case is presented in the output.

**Comment by kerrykimbrough** [comment, marked as answer]
You can't explicitly direct Tcases to generate a specific test case. But there is usually a way to get the test case you want by other means. For example, you can use properties and conditions to control which combinations of input are feasible. And you can use a [generator definition](http://www.cornutum.org/tcases/docs/Tcases-Guide.htm#higherCoverage) to ensure that all combinations of specific input variables are used.

**Comment by kerrykimbrough** [comment]
If you need to ensure that a specific _combination_ of inputs is covered, you can use the techniques described in the previous answer. But if you really need to ensure that a test case with a specific set of input values is covered, then Tcases is not the tool for the job. The goal of Tcases is rigorously cover the entire input space of the system-under-test, perhaps using input combinations that you might not normally consider.

**Comment by rahul15sel** [comment]
Where to place this Generators.xml file? File name should be specific to schema name or project name? I tried both way but seems its not considering file at all.

**Comment by kerrykimbrough** [comment]
Assuming that your input definition is a file named _projectName_-Input.xml, then by default generators are expected in a file named _projectName_-Generators.xml in the same directory. But you can override that at the command line using the `-g` option (or if using Maven, the `-DgenDef` option).

**Reply by rahul15sel** [reply]
Actually I am writing API test cases and my input definition is a swagger in .yaml file format under openapi folder.  I tried by placing projectname-Generators.xml in same folder and executed through command mvn tcases:api-test  -pl  :projectname but observed its not considering Generators.xml file as I tried with diff combination but always result is same.

Through documents I found that if there is no Generators.xml file then it generates itself but that option also did not work i.e. I removed Generators.xml file from project and tried command mvn tcases:api-test -r 12123322155 -pl :projectname.

Can you help here , appreciate your response.

**Reply by kerrykimbrough** [reply]
Sorry, the current version of Tcases for OpenAPI does not support generators for API tests. All API tests are created with "each choice" coverage only.
